utils/tfw_logger.cc

```diff
@@ -233,67 +268,76 @@ callback(const char *data, int size, void *private_data)
 }
 
 void
-run_thread(const int ncpu, const int fd, const std::string &host,
-	   const std::string &user, const std::string &password)
+run_thread(const int ncpu, const int fd, const TfwLoggerConfig &config) noexcept
 {
-	static thread_local std::chrono::seconds timeout(reconnect_min_timeout);
-	cpu_set_t cpuset;
-	pthread_t current_thread = pthread_self();
-	bool affinity_is_set = false;
-	int r;
-
-	while (!stop_flag) {
-		try {
-			TfwClickhouse clickhouse(host, table_name,
-						 user, password, make_block());
-
-			TfwMmapBufferReader mbr(ncpu, fd, &clickhouse, callback);
-
-			if (!affinity_is_set) {
-				CPU_ZERO(&cpuset);
-				CPU_SET(mbr.get_cpu_id(), &cpuset);
-
-				r = pthread_setaffinity_np(current_thread,
-							sizeof(cpu_set_t), &cpuset);
-				if (r != 0)
-					throw Except("Affinity setting failed");
-
-				affinity_is_set = true;
-			}
-
-			mbr.run(&stop_flag);
-
-			break;
-		}
-		catch (const Exception &e) {
-			log_error(e.what(), true, false);
-			/* All the manually thrown exceptions are critical */
-			break;
-		}
-		catch (const std::exception &e) {
-			if (!uncritical_error) {
-				log_error(e.what(), true, true);
-				timeout = reconnect_min_timeout;
-				uncritical_error = true;
+	try {
+		static thread_local std::chrono::seconds timeout(reconnect_min_timeout);
+		cpu_set_t cpuset;
+		pthread_t current_thread = pthread_self();
+		bool affinity_is_set = false;
+		int r;
+		
+		while (!stop_flag) {
+			try {
+				const auto &ch_cfg = config.get_clickhouse();
+				spdlog::debug("Worker {} connecting to ClickHouse "
+					      "at {}:{}, table: {}",
+					      ncpu,
+					      ch_cfg.host,
+					      ch_cfg.port,
+					      ch_cfg.table_name);
```

Comment:
There is no need to place every parameter on a single line. 
```suggestion
                                              ncpu, ch_cfg.host, ch_cfg.port,
                                              ch_cfg.table_name);
```

Also please use `try {} catch () {}` instead of `{}` for the function, reduce the nesting and adjust the code. The point is in reducing nesting, now writing 10 characters lines of code.

Please adjust the rest of the function appropriately